Secret-agency-service/src/main/java/cz/fi/muni/pa165/secretagency/service/GenericServiceImpl.java

```diff
@@ -0,0 +1,67 @@
+package cz.fi.muni.pa165.secretagency.service;
+
+import cz.fi.muni.pa165.secretagency.dao.GenericDao;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+/**
+ * Implementation of generic service. It provides general method implementations.
+ * Every other Service implementation should extend this one.
+ *
+ * @param <Entity> Entity
+ *
+ * 2author Jan Pavlu (487548)
+ */
+@Service
+public class GenericServiceImpl<Entity> implements GenericService<Entity> {
+
+    @Autowired
+    private GenericDao<Entity> genericDao;
```

Comment:
Jak by sis pak predstavoval tu implementujici servicu. Bude mit dalsi dao? Takze kazda servica bude mit 2 dao? Nebo se to nejak overridne?

Reply:
Good point, nad tím jsem nepřemýšlel. Třeba to takhle nepůjde. Asi by nebylo dobrý mít 2 dao pro jednu Service. Zamyslím se a pak to zahodím. 😄

Reply:
mrkni na ten muj PR, ja jsem to tam udelal pres consturctor a jede to i v testech

Reply:
Díval jsem se. Ta tvoje implementace je rozhodně použitelná. :) Jen jsem se snažil vyhnout nutnosti psát v každém z potomků konstruktor, protože si myslím, že Spring by mohl zvládnout injektovat správný repository automaticky. :) 
Ale vzhledem k tomu, že zatím nemám testy, ani konkrétní Service, tak to zatím nemůžu ověřit. 😄